plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/index/HudiColumnStatsIndexSupport.java

```diff
@@ -75,12 +76,12 @@ public class HudiColumnStatsIndexSupport
     private final Duration columnStatsWaitTimeout;
     private final long futureStartTimeMs;
 
-    public HudiColumnStatsIndexSupport(ConnectorSession session, SchemaTableName schemaTableName, Lazy<HoodieTableMetaClient> lazyMetaClient, Lazy<HoodieTableMetadata> lazyTableMetadata, TupleDomain<HiveColumnHandle> regularColumnPredicates)
+    public HudiColumnStatsIndexSupport(ConnectorSession session, SchemaTableName schemaTableName, Lazy<HoodieTableMetaClient> lazyMetaClient, Lazy<HoodieTableMetadata> lazyTableMetadata, TupleDomain<HiveColumnHandle> regularColumnPredicates, ExecutorService executor)
     {
-        this(log, session, schemaTableName, lazyMetaClient, lazyTableMetadata, regularColumnPredicates);
+        this(log, session, schemaTableName, lazyMetaClient, lazyTableMetadata, regularColumnPredicates, executor);
     }
 
-    public HudiColumnStatsIndexSupport(Logger log, ConnectorSession session, SchemaTableName schemaTableName, Lazy<HoodieTableMetaClient> lazyMetaClient, Lazy<HoodieTableMetadata> lazyTableMetadata, TupleDomain<HiveColumnHandle> regularColumnPredicates)
+    public HudiColumnStatsIndexSupport(Logger log, ConnectorSession session, SchemaTableName schemaTableName, Lazy<HoodieTableMetaClient> lazyMetaClient, Lazy<HoodieTableMetadata> lazyTableMetadata, TupleDomain<HiveColumnHandle> regularColumnPredicates, ExecutorService executor)
```

Comment:
Given all indexes are reusing the same executor service, should the default parallelism `hudi.split-generator-parallelism` controlling the pool size be increased and config doc be updated?